java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -503,10 +561,26 @@ public void visitNewClass(NewClassTree tree) {
 
   private JavaSymbol resolveConstructorSymbol(IdentifierTree identifier, Type type, Resolve.Env methodEnv, List<JavaType> argTypes) {
     JavaSymbol symbol = resolve.findMethod(methodEnv, (JavaType) type, "<init>", argTypes).symbol();
+    inferArgumentTypes(argTypes, symbol);
     associateReference(identifier, symbol);
     return symbol;
   }
 
+  private void inferArgumentTypes(List<JavaType> argTypes, JavaSymbol symbol) {
+    Type formal = Symbols.unknownType;
+    for (int i = 0; i < argTypes.size(); i++) {
+      if (symbol.isMethodSymbol()) {
+        Symbol.MethodSymbol methodSymbol = (Symbol.MethodSymbol) symbol;
+        int size = methodSymbol.parameterTypes().size();
+        formal = methodSymbol.parameterTypes().get(i < size ? i : size - 1);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS864)